### DISCUSSION TITLE
Zoom & pan feature

### DISCUSSION THREAD

**Discussion by SizzlingBytes** [recharts/recharts#3452] [Q&A]
How come recharts doesn't support zoom and pan features? any plans to build them?

**Comment by ckifer** [comment]
Hi, please see this example
https://recharts.org/en-US/examples/HighlightAndZoomLineChart

**Reply by SizzlingBytes** [reply]
thanks for the response. i was thinking of something like this 
![zoom and pan](https://js.devexpress.com/Content/images/doc/22_2/ChartJS/devextreme-chart-NewZoomScroll.gif)

ability to zoom in & out using the mouse wheel
scroll the graph left and right with the mouse drag

**Reply by ckifer** [reply]
something like this probably could be hacked together in your application, but there isn't a way to do it out of the box. 

See the comment on this issue https://github.com/recharts/recharts/issues/710#issuecomment-1312727056

There currently isn't plans for it because we're trying to aim for better docs and stability as a repo before adding any new large features like this one. I did re-open that issue though

**Reply by KushalTanna24** [reply]
++ this would be really helpful

**Reply by michelle-jca** [reply]
We would also love this feature! Especially the ability to scroll once you've zoomed in

**Comment by nhaebigkerber** [comment]
Just curious if this is on the roadmap anytime soon after the Big 3.0 upgrade?

**Reply by nhaebigkerber** [reply]
My organization is already committed to recharts and zooming is the next big push on our roadmap, so if this isn't planned for anything soon, I will build it myself and could share my implementation once complete if helpful.

**Reply by ckifer** [reply]
@nhaebigkerber it is not planned by us per se, but definitely welcome any additions in this realm of things. There was an attempt here https://github.com/recharts/recharts/pull/5934 but we had feedback/implementor ran out of time
